src/execution_tree/primitives/equal.cpp

```diff
@@ -55,24 +55,63 @@ namespace phylanx { namespace execution_tree { namespace primitives
             using operand_type = ir::node_data<double>;
             using operands_type = std::vector<primitive_argument_type>;
 
-            bool equal0d(operand_type&& lhs, operand_type&& rhs) const
+            primitive_argument_type equal0d1d(
+                operand_type&& lhs, operand_type&& rhs) const
+            {
+                // TODO: SIMD functionality should be added, blaze implementation
+                // is not currently available
+                rhs.vector() = blaze::map(rhs.vector(),
+                    [&](double x) { return (x == lhs.scalar()); });
+
+                return primitive_argument_type(ir::node_data<bool>{rhs});
+            }
+
+            primitive_argument_type equal0d2d(
+                operand_type&& lhs, operand_type&& rhs) const
+            {
+                // TODO: SIMD functionality should be added, blaze implementation
+                // is not currently available
+                rhs.matrix() = blaze::map(rhs.matrix(),
+                    [&](double x) { return (x == lhs.scalar()); });
+
+                return primitive_argument_type(ir::node_data<bool>{rhs});
+            }
+
+            primitive_argument_type equal0d(operand_type&& lhs, operand_type&& rhs) const
             {
                 std::size_t rhs_dims = rhs.num_dimensions();
                 switch(rhs_dims)
                 {
                 case 0:
-                    return lhs.scalar() == rhs.scalar();
+                    return primitive_argument_type(
+                        ir::node_data<bool>{lhs.scalar() == rhs.scalar()});
+
+                case 1:
+                    return equal0d1d(std::move(lhs), std::move(rhs));
+
+                case 2:
+                    return equal0d2d(std::move(lhs), std::move(rhs));
 
-                case 1: HPX_FALLTHROUGH;
-                case 2: HPX_FALLTHROUGH;
                 default:
                     HPX_THROW_EXCEPTION(hpx::bad_parameter,
                         "equal::equal0d",
                         "the operands have incompatible number of dimensions");
                 }
             }
 
-            bool equal1d1d(operand_type&& lhs, operand_type&& rhs) const
+            primitive_argument_type equal1d0d(
+                operand_type&& lhs, operand_type&& rhs) const
+            {
+                // TODO: SIMD functionality should be added, blaze implementation
+                // is not currently available
+                lhs.vector() = blaze::map(lhs.vector(),
+                    [&](double x) { return (x == rhs.scalar()); });
+
+                return primitive_argument_type(ir::node_data<bool>{lhs});
```

Comment:
You might want to consider moving lhs into the result:

     return primitive_argument_type(ir::node_data<bool>{std::move(lhs)});